x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -0,0 +1,169 @@
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xpack.application.connector.Connector;
+import org.elasticsearch.xpack.application.connector.ConnectorFiltering;
+import org.elasticsearch.xpack.application.connector.ConnectorIndexService;
+import org.elasticsearch.xpack.application.connector.ConnectorIngestPipeline;
+import org.elasticsearch.xpack.application.connector.ConnectorTemplateRegistry;
+import org.elasticsearch.xpack.application.connector.syncjob.action.PostConnectorSyncJobAction;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.xpack.core.ClientHelper.CONNECTORS_ORIGIN;
+
+/**
+ * A service that manages persistent {@link ConnectorSyncJob} configurations.
+ */
+public class ConnectorSyncJobIndexService {
+
+    private static final Long ZERO = 0L;
+
+    private final Client clientWithOrigin;
+
+    public static final String CONNECTOR_SYNC_JOB_INDEX_NAME = ConnectorTemplateRegistry.CONNECTOR_SYNC_JOBS_INDEX_NAME_PATTERN;
+
+    /**
+     * @param client A client for executing actions on the connectors sync jobs index.
+     */
+    public ConnectorSyncJobIndexService(Client client) {
+        this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
+    }
+
+    /**
+     * @param request   Request for creating a connector sync job.
+     * @param listener  Listener to respond to a successful response or an error.
+     */
+    public void createConnectorSyncJob(
+        PostConnectorSyncJobAction.Request request,
+        ActionListener<PostConnectorSyncJobAction.Response> listener
+    ) {
+        try {
+            getSyncJobConnectorInfo(request.getId(), new ActionListener<>() {
+                @Override
+                public void onResponse(Connector connector) {
+                    Instant now = Instant.now();
+                    ConnectorSyncJobType jobType = Objects.requireNonNullElse(request.getJobType(), ConnectorSyncJob.DEFAULT_JOB_TYPE);
+                    ConnectorSyncJobTriggerMethod triggerMethod = Objects.requireNonNullElse(
+                        request.getTriggerMethod(),
+                        ConnectorSyncJob.DEFAULT_TRIGGER_METHOD
+                    );
+
+                    try {
+                        String syncJobId = generateId();
+
+                        final IndexRequest indexRequest = new IndexRequest(CONNECTOR_SYNC_JOB_INDEX_NAME).id(syncJobId)
```

Comment:
```suggestion
                        final IndexRequest indexRequest = new IndexRequest(CONNECTOR_SYNC_JOB_INDEX_NAME).id(UUIDs.base64UUID())
```

this is what generating the ID does under the hood, I think it's good to use it here

Reply:
Adjusted with https://github.com/elastic/elasticsearch/pull/102714/commits/c21492cece4fa8e94c061f360e75fc0552cebb3a. I'm still keeping it inside the method as it explains the reasoning, why we do this "low-level" call by ourselves.